core/src/main/java/org/mule/runtime/core/internal/lifecycle/phases/MuleContextStopPhase.java

```diff
@@ -63,7 +68,19 @@ public MuleContextStopPhase(Class<?>[] ignorredObjects) {
         Stoppable.class
     });
 
-    setIgnoredObjectTypes(ignorredObjects);
+    if (parseBoolean(getProperty(REUSE_GLOBAL_ERROR_HANDLER_PROPERTY))) {
+      ignoredObjects = new Class[] {
```

Comment:
this should only add `GlobalErrorHandler.class` to the passed array, otherwise this would break if the given collection is modified in the future.

Reply:
fixed